run/run.go

```diff
@@ -12,39 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package sdk
+package run
```

Comment:
Is there a reason this is in `run`? I'm thinking we could move it to the root so it's easier to be called (`sdk.Run`) and add a `go:build wasm` constraint, so it can only be used in the entrypoint of the processor if it also contains the same constraint. This gives us the nice side-effect that users can't compile the processor into anything other than a WASM binary, it guides towards doing the right thing.

Reply:
Yeah, I started with `sdk.Run()`. However, I left it because of an import cycle. `Run()` uses `NextCommand()` from internal, which in turn uses `sdk`. If we put `Run()` into `sdk`, we'd get that import cycle.

Adding `go:build wasm` sounds good!

Reply:
I moved the entrypoint to the root now. This meant moving command actions there as well, as they are the ones causing the circular dependency otherwise. But it's all hidden away from the user and we hopefully don't have to touch it again, at least not soon.